genmc-sys/src/genmc.rs

```diff
@@ -0,0 +1,39 @@
+use std::sync::OnceLock;
+
+use cxx::UniquePtr;
+
+use super::{GenmcParams, LogLevel, MiriGenmcShim};
+
+static GENMC_LOG_LEVEL: OnceLock<LogLevel> = OnceLock::new();
+
+#[derive(Clone, Copy)]
+/// This struct is used to prevent unsafe access to GenMC.
+/// Any operations that create a GenMC object or call a GenMC function should go through this struct.
+pub struct Genmc {}
```

Comment:
Why does this need a separate type? It could be part of creating a `MiriGenmcShim`.